packages/components/src/TabBar/TabBar.component.js

```diff
@@ -102,7 +105,7 @@ TabBar.propTypes = {
 		}).isRequired,
 	).isRequired,
 	onSelect: PropTypes.func.isRequired,
-	selectedKey: PropTypes.string,
+	selectedKey: PropTypes.any,
```

Comment:
So the component will work evenly wether with a string/number/object/array/boolean ?

Reply:
This is passed to react-bootstrap tabs, which propType for this props is any. I guess they shallow compare them, so any key with the same value/reference will work.